### PR TITLE
GRID-431: Misaligned filter cell editor

### PR DESCRIPTION
**Pull request by joneit**
See Jira [GRID-431](https://appoji.jira.com/browse/GRID-431) for details.

### CODE REVIEW

File: src/Hypergrid.js [hunk 1]
```diff
@@ -1075,7 +1073,7 @@ var Hypergrid = Base.extend('Hypergrid', {
             event.isDataColumn &&
             event.getCellProperty(event.isDataRow ? 'editable' : 'filterable')
         ) {
-            this.setMouseDown(event.gridCell);
+            this.setMouseDown(new Point(event.gridCell.x, event.dataCell.y));
```
Comment (Dwaynekj) [inline src/Hypergrid.js]: they take Points now?

File: src/cellEditors/CellEditor.js [hunk 2]
```diff
@@ -242,8 +233,6 @@ var CellEditor = Base.extend('CellEditor', {
             this.grid.cellEditor = null;
             this.el.remove();
         } else if (feedback >= 0) { // never true when `feedback` undefined
-            var point = this.event.gridCell;
-            this.grid.selectViewportCell(point.x, point.y - this.grid.getHeaderRowCount());
```
Comment (Dwaynekj) [inline src/cellEditors/CellEditor.js]: This removes the selection?

File: src/defaults.js [hunk 3]
```diff
@@ -653,6 +653,17 @@ var defaults = {
...
     /**
+     * This is used only by FilterBox cell editor.
+     * One of:
+     * * **`'onCommit'`** - Column filter state not set until keyup === `\r` (return/enter key)
+     * * **`'immediate'`** - Column filter state set on each key press
+     * @default
+     * @type {boolean}
+     * @memberOf module:defaults
+     */
+    filteringMode: 'onCommit',
```
Comment (Dwaynekj) [inline src/defaults.js]: My instinctive reaction is that filteringMode sounds very application layer. Our grid really is unawares of any notion of filtering

File: src/Hypergrid.js (hunk 1 above)
Comment (joneit) [inline src/Hypergrid.js]: always did, i believe

File: src/cellEditors/CellEditor.js (hunk 2 above)
Comment (joneit) [inline src/cellEditors/CellEditor.js]: not sure why i removed this.... the header row count was probably out of date. i'll revisit.

File: src/defaults.js (hunk 3 above)
Comment (Dwaynekj) [inline src/defaults.js]: Ah ok

File: src/cellEditors/CellEditor.js (hunk 2 above)
Comment (Dwaynekj) [inline src/cellEditors/CellEditor.js]: Ok. So wait to thumbs?

Comment (joneit) [inline src/cellEditors/CellEditor.js]: The intent of the deleted line in question is to reposition the edit box before giving an error animation + alert. Not sure why that was necessary but seems to work without it. More to the point, that function `selectViewportCell ` is broken and anyway and needs to be fixed. We can do that later. This should be good to go as is.

**Pull request merged**